### PR TITLE
feat: fix linter errors and add wait to machinepool spec

### LINKED ISSUE #6: fix: tests
Current tests are failing.  Need to fix these.

### PR DESCRIPTION
**Pull request by scottd018**
- Fixes #6

### CODE REVIEW

File: api/v1alpha1/gitlabidentityprovider_types.go [hunk 1]
```diff
@@ -117,7 +119,7 @@ type GitLabIdentityProvider struct {
 
 //+kubebuilder:object:root=true
 
-// GitLabIdentityProviderList contains a list of GitLabIdentityProvider
+// GitLabIdentityProviderList contains a list of GitLabIdentityProvider.
```
Comment (daxelrod-rh) [inline api/v1alpha1/gitlabidentityprovider_types.go:122]: Not a blocker, but why is a comment like this useful at all?

File: controllers/gitlabidentityprovider/controller.go [hunk 2]
```diff
@@ -107,6 +107,8 @@ func (r *Controller) ReconcileDelete(req controllers.Request) (ctrl.Result, erro
 }
 
 // SetupWithManager sets up the controller with the Manager.
+//
+//nolint:wrapcheck
```
Comment (daxelrod-rh) [inline controllers/gitlabidentityprovider/controller.go:111]: Why are we disabling wrapcheck here?

File: controllers/ldapidentityprovider/controller.go [hunk 3]
```diff
@@ -102,6 +102,8 @@ func (r *Controller) ReconcileDelete(req controllers.Request) (ctrl.Result, erro
 }
 
 // SetupWithManager sets up the controller with the Manager.
+//
+//nolint:wrapcheck
```
Comment (daxelrod-rh) [inline controllers/ldapidentityprovider/controller.go:106]: Ditto for this one; why disable wrapcheck?

**Review by daxelrod-rh — CHANGES_REQUESTED** [review]
This all looks good, except I'm just curious about a couple of places where we're disabling wrapcheck without it being obvious to me why we would. It might be worth explicitly documenting the reason for functions where we disable it.

File: api/v1alpha1/gitlabidentityprovider_types.go (hunk 1 above)
Comment (scottd018) [reply api/v1alpha1/gitlabidentityprovider_types.go:122]: This is scaffolded from Kubebuilder.  Easier to leave them in place than remove them.

File: controllers/gitlabidentityprovider/controller.go (hunk 2 above)
Comment (scottd018) [reply controllers/gitlabidentityprovider/controller.go:111]: This is scaffolded from Kubebuilder.  Wrapcheck is a pretty opinionated check IMO.  I disable this frequently when wrapping errors doesn't make sense.

File: controllers/ldapidentityprovider/controller.go (hunk 3 above)
Comment (scottd018) [reply controllers/ldapidentityprovider/controller.go:106]: See above.  Same reason.

**Review by daxelrod-rh — APPROVED** [review]
:shipit:

**Pull request merged**